lib/tasks/resque.rake

```diff
@@ -50,10 +52,31 @@ namespace :resque do
     if pids.empty?
       puts "No workers to kill"
     else
+      # First tell workers to stop accepting new work by sending USR2 signal
+      puts "\nTelling workers to finish current jobs, but not process any new jobs..."
+      syscmd = "kill -s USR2 #{pids.join(' ')}"
+      puts "$ #{syscmd}"
+      `#{syscmd}`
+      puts "\n"
+      puts "Waiting for workers to finish current jobs..."
+      start_time = Time.now
```

Comment:
It probably won't matter too much in this case because you are consistently using `Time.now` but for rails app `Time.current` is better because it takes into consideration the timezone set by the rails application.